lib/pnode/server.ml

```diff
@@ -52,13 +53,15 @@ let process_message node preprocessor msg_handler =
               !node.address.address !node.address.port message.sender.address
               message.sender.port) in *)
       match msg_handler message with
-      | Some response ->
-        response
-        |> Client.create_response node message
+      | Some msg ->
```

Comment:
We can destructure here (just a matter of syntax); up to you
```ocaml
| Some Message.{ data ; signature = payload_signature }
  Client.create_response node message ?payload_signature data

Reply:
Nice one, totally missed it. Done.